### PR TITLE
Support API to prepare a workspace for e2e tests

### PR DESCRIPTION
**Pull request by ashwin1111**

### CODE REVIEW

File: sql/functions/reset-workspace.sql [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+  GET DIAGNOSTICS rcount = ROW_COUNT;
+  RAISE NOTICE 'Deleted % workspace_schedules', rcount;
+
+  DELETE
+  FROM django_q_schedule dqs
+  WHERE dqs.args = _workspace_id::varchar(255);
+  GET DIAGNOSTICS rcount = ROW_COUNT;
+  RAISE NOTICE 'Deleted % django_q_schedule', rcount;
+
+--   DELETE
```
Comment (ashwin1111) [inline sql/functions/reset-workspace.sql:243]: Kept these lines commented for better visibility

File: fyle_qbo_api/settings.py [hunk 2]
```diff
@@ -261,6 +261,9 @@
 QBO_REDIRECT_URI = os.environ.get('QBO_REDIRECT_URI')
 QBO_TOKEN_URI = os.environ.get('QBO_TOKEN_URI')
 QBO_ENVIRONMENT = os.environ.get('QBO_ENVIRONMENT')
+ENCRYPTION_KEY = os.environ.get('ENCRYPTION_KEY')
```
Comment (ashwin1111) [inline fyle_qbo_api/settings.py:264]: TODO: Add these secrets to fyle-deploy

File: tests/test_workspaces/test_views.py [hunk 3]
```diff
@@ -195,3 +195,24 @@ def test_post_connect_qbo_view(api_client, test_connection):
...
+        'setup-e2e-test', kwargs={
+            'workspace_id': 1
+        }
+    )
+
+    api_client.credentials(HTTP_X_E2E_Tests_Client_ID='dummy_id')
+    response = api_client.post(url)
+    assert response.status_code == 403
+
+    api_client.credentials(HTTP_X_E2E_Tests_Client_ID='gAAAAABi8oXHBll3lEUPGpMDXnZDhVgSl_LMOkIF0ilfmSCL3wFxZnoTIbpdzwPoOFzS0vFO4qaX51JtAqCG2RBHZaf1e98hug==')
```
Comment (ashwin1111) [inline tests/test_workspaces/test_views.py:212]: both the keys are generated against a different key, so safe to keep them hardcoded in the repo

Comment (ashwin1111) [reply tests/test_workspaces/test_views.py:212]: the key is used only for CI run

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**